src/amberscript/parser.cc

```diff
@@ -96,6 +90,7 @@ Result Parser::Parse(const std::string& data) {
     if (!r.IsSuccess())
       return Result(make_error(r.Error()));
   }
+  script_->SetCommands(std::move(command_list_));
```

Comment:
Maybe it's a subtle point, but this actually does ensure that command_list is empty afterward.
The std::vector move constructor guarantees it.

Reply:
Yes, but I need to swap it back to get the previous command list entries back into the list.